packages/api/src/tasks/checkAlerts.ts

```diff
@@ -160,27 +160,144 @@ export const notifyChannel = async ({
             }),
       });
 
-      // ONLY SUPPORTS SLACK WEBHOOKS FOR NOW
       if (webhook?.service === 'slack') {
-        await slack.postMessageToWebhook(webhook.url, {
-          text: message.title,
-          blocks: [
-            {
-              type: 'section',
-              text: {
-                type: 'mrkdwn',
-                text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
-              },
-            },
-          ],
-        });
+        await handleSendSlackWebhook(webhook, message);
+      } else if (webhook?.service === 'generic') {
+        await handleSendGenericWebhook(webhook, message);
       }
       break;
     }
     default:
       throw new Error(`Unsupported channel type: ${channel}`);
   }
 };
+
+const handleSendSlackWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  if (!webhook.url) {
+    throw new Error('Webhook URL is not set');
+  }
+
+  await slack.postMessageToWebhook(webhook.url, {
+    text: message.title,
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
+        },
+      },
+    ],
+  });
+};
+
+export const handleSendGenericWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  // QUERY PARAMS
+
+  if (!webhook.url) {
+    throw new Error('Webhook URL is not set');
+  }
+
+  let url: string;
+  // user input of queryParams is disabled on the frontend for now
+  if (webhook.queryParams) {
+    const queryParamsString = new URLSearchParams(
+      webhook.queryParams,
+    ).toString();
+
+    // user may have included params in both the url and the query params
+    // so they should be merged
+    const tmpURL = new URL(webhook.url);
+    if (queryParamsString) {
+      tmpURL.search = tmpURL.search
+        ? `${tmpURL.search}&${queryParamsString}`
+        : queryParamsString;
+    }
+    url = tmpURL.toString();
+  } else {
+    // if there are no query params given, just use the url
+    url = webhook.url;
+  }
+
+  // HEADERS
+  // TODO: handle real webhook security and signage after v0
+  // X-HyperDX-Signature FROM PRIVATE SHA-256 HMAC, time based nonces, caching functionality etc
+
+  const headers = {
+    'Content-Type': 'application/json', // default, will be overwritten if user has set otherwise
+    ...(webhook.headers?.toObject() || {}),
```

Comment:
nit: `??` is probably what we want here ?

Reply:
oh yeah this is an artifact too mb, when we started with strings and they could be empty strings it needed the falsy but that isn't needed - changing now